pushy/src/main/java/com/eatthepath/pushy/apns/ApnsClientHandler.java

```diff
@@ -332,7 +328,16 @@ private void handleEndOfStream(final ChannelHandlerContext context, final Http2S
                     true, getApnsIdFromHeaders(headers), null, null));
         } else {
             if (data != null) {
-                final ErrorResponse errorResponse = GSON.fromJson(data.toString(StandardCharsets.UTF_8), ErrorResponse.class);
+                ErrorResponse errorResponse;
+
+                try {
+                    errorResponse = ErrorResponse.fromMap(
+                            jsonDeserializer.parseJsonObject(data.toString(StandardCharsets.UTF_8)));
+                } catch (final ParseException e) {
+                    log.error("Failed to parse error reasponse: data.toString(StandardCharsets.UTF_8)");
+                    errorResponse = new ErrorResponse(null, null);
```

Comment:
Oops—meant to leave a TODO here to revisit this and make sure it's really the safest approach.

Reply:
Having fixed the glaring typos, I think this is fine. This is exactly what would have happened before, but now we have a logging statement to warn us if it ever happens.